src/mlpack/tests/ann_layer_test.cpp

```diff
@@ -5664,6 +5718,91 @@ TEST_CASE("NoBiasConvolutionLayerTestCase", "[ANNLayerTest]")
   REQUIRE(arma::accu(output) == 4108);
 }
 
+/**
+ * Simple test for the Grouped Convolution layer.
+ */
+TEST_CASE("GroupedConvolutionLayerTest", "[ANNLayerTest]")
+{
+  arma::mat input, output;
+
+  // The input test matrix is of the form 3 x 2 x 4 x 1 where
+  // number of images are 3 and number of feature maps are 2.
+  input = { { 1, 446, 42 },
+            { 2, 16, 63 },
+            { 3, 13, 63 },
+            { 4, 21, 21 },
+            { 1, 13, 11 },
+            { 32, 45, 42 },
+            { 22, 16 , 63 },
+            { 32, 13 , 42 } };
+
+  GroupedConvolution layer(4, 1, 1, 2, 1, 1, 0, 0);
+  layer.InputDimensions() = std::vector<size_t>({ 4, 1, 2 });
+  layer.ComputeOutputDimensions();
+  arma::mat layerWeights(layer.WeightSize(), 1);
+  layerWeights(0) = -0.4886662960;
+  layerWeights(1) = -0.4388893843;
+  layerWeights(2) = -0.7269541025;
+  layerWeights(3) = 0.7835500240;
+  layerWeights(4) = -0.6865804195;
+  layerWeights(5) = -0.7586858273;
+  layerWeights(6) = -0.1721059084;
+  layerWeights(7) = -0.1972532272;
+  layer.SetWeights(layerWeights.memptr());
+  output.set_size(layer.OutputSize(), 3);
+
+  layer.Forward(input, output);
+
+  // Value calculated using torch.nn.Conv2d().
+  REQUIRE(arma::accu(output) == Approx(-647.6369628906).epsilon(1e-5));
```

Comment:
Another option here, if this really is just one group, would be to test that regular `Convolution<>` gets the same result.

Reply:
Nope. It uses 2 groups.